### PR TITLE
[Merged by Bors] - Added a new sync stress test

### PR DESCRIPTION
**Pull request by AmitShaul**
## Motivation
We want to check our prod is working well under stress

### CODE REVIEW

File: tests/stress/sync_stress/test_sync.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    max_retries = 20
+    interval_time = 60
+    print("waiting for new pod to be synced")
+    while True:
+        hits = q.get_all_msg_containing(init_session, new_client, SYNC_DONE, is_print=False)
+        if hits:
+            break
+
+        print(f"not synced after {curr_try} tries of {interval_time} secs each", end="\r")
+        time.sleep(interval_time)
```
Comment (antonlerner) [inline tests/stress/sync_stress/test_sync.py:65]: sleep for 1 minute?

File: tests/stress/sync_stress/test_sync.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    del cspec.args['remote-data']
+    cspec.args['data-folder'] = ""
+
+    # Adding a new client
+    res_lst = add_multi_clients(init_session, cspec, 1, 'client')
+    new_client = res_lst[0]
+
+    curr_try = 0
+    max_retries = 20
```
Comment (antonlerner) [inline tests/stress/sync_stress/test_sync.py]: please comment on the sizes and measures, why are we waiting for this long? shouldn't it be a function of the sample size?

File: tests/stress/sync_stress/test_sync.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        curr_try += 1
+
+    # parsing sync start time
+    start_sync_hits = q.get_all_msg_containing(init_session, new_client, START_SYNC, is_print=False)
+    last_sync_msg = start_sync_hits[-1]
+    st = convert_ts_to_datetime(last_sync_msg["T"])
+    et = convert_ts_to_datetime(hits[0]["T"])
+    # total time since starting sync until finishing
+    print(f"new client is synced after {str(et - st)}")
```
Comment (antonlerner) [inline tests/stress/sync_stress/test_sync.py:84]: assert that sync time is less than some threshold

File: tests/stress/sync_stress/test_sync.py (hunk 2 above)
Comment (antonlerner) [reply tests/stress/sync_stress/test_sync.py]: also perhaps you can print the sample size?

File: tests/stress/sync_stress/test_sync.py (hunk 1 above)
Comment (AmitShaul) [reply tests/stress/sync_stress/test_sync.py:65]: yes.. do you think it should be a diff value?

File: tests/stress/sync_stress/test_sync.py (hunk 2 above)
Comment (AmitShaul) [reply tests/stress/sync_stress/test_sync.py]: I don't have the sample size in my code, it's transparent to the test so either way we'll have to change it manually with every sample swap.

File: tests/stress/sync_stress/test_sync.py (hunk 3 above)
Comment (AmitShaul) [reply tests/stress/sync_stress/test_sync.py:84]: assert curr_try < max_retries -> this assertion is making sure we're not exceeding here.
I can add another assertion though

**Review by almogdepaz — APPROVED** [review]

File: tests/stress/sync_stress/test_sync.py (hunk 3 above)
Comment (antonlerner) [reply tests/stress/sync_stress/test_sync.py:84]: This is a differrent assert, one is for having the correct logic, and the other is for doing the sync in a designated time frame. we need to check both

File: tests/stress/sync_stress/test_sync.py (hunk 1 above)
Comment (antonlerner) [reply tests/stress/sync_stress/test_sync.py:65]: It depends on how long the entire sync is done, if it takes 5 sec I wouldn't wait a minute vefore polling again

File: tests/stress/sync_stress/test_sync.py (hunk 3 above)
Comment (AmitShaul) [reply tests/stress/sync_stress/test_sync.py:84]: K adding it

File: tests/stress/sync_stress/test_sync.py (hunk 2 above)
Comment (antonlerner) [reply tests/stress/sync_stress/test_sync.py]: ok just leave a comment for which sample is used and whats its size... other wise we wouldn't know

File: tests/stress/sync_stress/test_sync.py (hunk 1 above)
Comment (AmitShaul) [reply tests/stress/sync_stress/test_sync.py:65]: it takes between 12-18 mins the whole syncing process.
sleeping for 1 min shows process as well. it doesn't look stuck.
not critical to me. I can change it to 2 mins if you'd like

File: tests/stress/sync_stress/test_sync.py (hunk 2 above)
Comment (AmitShaul) [reply tests/stress/sync_stress/test_sync.py]: you can see it in the conf yaml file, I named the data folder as the number of days you recorded.
i'll add a comment as well

**Review by antonlerner — APPROVED** [review]

**Pull request closed without merging**